x/programs/wasmlanche/src/context.rs

```diff
@@ -350,6 +340,17 @@ impl Context {
             Context::External(ctx) => ctx.call_function(function_name, args),
         }
     }
+
+    #[cfg(feature = "bindings")]
+    #[must_use]
+    pub fn new_external_call_context(&self, address: Address, max_units: Gas, value: u64) -> Self {
+        Self::External(ExternalCallContext::new(
+            self.host_accessor().clone(),
```

Comment:
cloning is actually fine here since `ExternalCallContext` can only call into `call_function` and won't have access to state in unit tests

Reply:
As discussed offline, `ExternalCallContext` should take a mutable borrow to `Context` instead of being part of the enum. Then we can write a wrapper function for `bindings` compilations that takes the `ExternalCallContext` but uses the mutable borrow of `Context` to actually perform the function call. 

I'll start working on that after this lands.